### DISCUSSION TITLE
exportclone tsv not txt？vdjtools can‘t convert

### DISCUSSION THREAD

**Discussion by GANgan2022** [milaboratory/mixcr#1128] [Q&A] [answered]
The output of mixcr exportclone is in TSV format, separated by tabs. However, when I try to convert the TSV results into vdjtools-specific format using the vdjtools command (java -jar /home/cyf/app/vdjtools-1.2.1/vdjtools-1.2.1.jar convert -S mixcr output2/*.tsv outvdj/), I encounter various formatting issues that prevent the conversion. For example, the header needs to be changed from "read count" to "clonecount", and I need to remove some extra commas. Additionally, when converting BCR-seq data, I receive errors such as "MiXcr input type: Unknown symbol 'e'", and "Some mandatory columns are absent in the input file."

**Comment by mizraelson** [comment]
Hi,

vdjtools is a legacy software from our suite and is no longer actively supported. As a result, converting current MiXCR output to be compatible with vdjtools must be done manually. However, it's worth noting that many of the features originally present in VDJtools have since been incorporated directly into MiXCR.

**Comment by jialin-li-cell** [comment]
emergency!!!，could anyone teach me how to solve this problem? 
 
 ljl@Jerry:~/vdjtools-1.2.1$ java --illegal-access=deny --add-opens java.base/java.lang=ALL-UNNAMED -jar vdjtools-1.2.1.jar Convert -S mixcr /home/ljl/A2-8result/A2-8result.clones_TRB.tsv   /home/ljl/A2-8result/A2-8VDJtools
Executing com.antigenomics.vdjtools.misc.Convert -S mixcr /home/ljl/A2-8result/A2-8result.clones_TRB.tsv /home/ljl/A2-8result/A2-8VDJtools
[Sun Aug 11 16:13:01 CST 2024 Convert] Reading sample(s)
[Sun Aug 11 16:13:01 CST 2024 Convert] 1 sample(s) loaded
[Sun Aug 11 16:13:01 CST 2024 SampleStreamConnection] Loading sample A2-8result.clones_TRB
[ERROR] java.lang.RuntimeException: Unable to parse clonotype string 0  1953.0  3.084553729988306E-4    TGCAGTGCTAGAGATGTAGACTCTGGGACCCAGTACTTC [[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[ TRBV20-1*00(904.6)              TRBJ2-5*00(406.7)      679|694|713|0|15||150.0          27|40|68|26|39||130.0           TGCAGTGCTAGAGATGTAGACTCTGGGACCCAGTACTTC 58      CSARDVDSGTQYF   :::::::::0:1:15:::::26:-7:39::: for MiXcr input type: Some mandatory columns are absent in the input file., see _vdjtools_error.log for details

could anyone please teach me how to solve this?thanks!!!

**Reply by mizraelson** [reply, marked as answer]
Hi, you can modify MiXCR output manually to work with VDJTools using the method described here: #1617
